### ISSUE TITLE
[0.49.7] [Stratcon] Recon lances are immediately engaged even when scenarios claim to be in the future

### ISSUE THREAD

**Issue by mjbroekman** [MegaMek/mekhq#3382]
### Environment

- MekHQ: 0.49.7
- Windows 10
- Java Vendor: Eclipse Adoptium
- Java Version: 11.0.13

### Description
When I land on a planet and deploy lances to recon the Stratcon maps, if they land on an encounter as these two below:
![Screenshot 2022-06-26 00 15 20](https://user-images.githubusercontent.com/5141448/175799160-4af14646-2ce8-496f-8de3-f1819a75b7e0.png)

The Command Center tab / Daily Report shows the encounters for different days in the future:
![Screenshot 2022-06-26 00 15 56](https://user-images.githubusercontent.com/5141448/175799170-29978aeb-9f22-4672-a8df-ff68825510d0.png)

However, the Briefing Room sets the date to the current date:
![Screenshot 2022-06-26 00 16 21](https://user-images.githubusercontent.com/5141448/175799177-b03de258-c990-4c7f-8973-9090ef68591d.png)

### Recommendation
I'm not sure which one is the "correct" one, but it feels like the Daily Report should be the correct one as it would account for travel time for the lance to _reach_ the destination map hex let alone conduct "recon".

Either way, the Briefing Room and the Daily Report / Command Center need to be in sync

**Comment by IllianiCBT** [comment]
Would you be able to attach a copy of your save, including any custom units. If possible please include a before & after save. I’ll do some testing tomorrow evening

**Comment by mjbroekman** [comment]
Unfortunately, I only have the "after" save. The only "before" save I have is 18 months prior during a break between contracts and performing some massive refits.
"After" save: [Star League Reborn30330418.cpnx.gz](https://github.com/MegaMek/mekhq/files/8993040/Star.League.Reborn30330418.cpnx.gz)
Userdata folder with custom units: [userdata.zip](https://github.com/MegaMek/mekhq/files/8993042/userdata.zip)

**Comment by mjbroekman** [comment]
Different contract: "Before" save: [Autosave-1-Star League Reborn-30340617.cpnx.gz](https://github.com/MegaMek/mekhq/files/9006217/Autosave-1-Star.League.Reborn-30340617.cpnx.gz)

"After" save:
[Star League Reborn30340619.cpnx.gz](https://github.com/MegaMek/mekhq/files/9006224/Star.League.Reborn30340619.cpnx.gz)

Action: Recon @ Track 0 - 8,5

Daily Report:
New scenario "Prevent Breakthrough" will occur on 3034-06-23.

Briefing Room:
Prevent Breakthrough - 3034-06-19

AtB Campaign State:
Deployment Date: 3034-06-19
Battle Date: 3034-06-19

**Comment by IllianiCBT** [comment]
Thank you. I’ll take a look when I have a moment tomorrow :)

**Comment by IllianiCBT** [comment]
Throwing the pre-issue save into .7 and I was able to replicate this issue without problem

From the log:
`08:13:24,242 WARN  [mekhq.campaign.mission.atb.AtBScenarioManifest] {AWT-EventQueue-0}
mekhq.campaign.mission.atb.AtBScenarioManifest.Deserialize(AtBScenarioManifest.java:48) - Specified file ./data/scenariotemplates/UserScenarioManifest.xml does not exist

08:13:24,373 WARN  [mekhq.campaign.mission.atb.ScenarioModifierManifest] {AWT-EventQueue-0}
mekhq.campaign.mission.atb.ScenarioModifierManifest.Deserialize(AtBScenarioModifier.java:625) - Specified file ./data/scenariomodifiers/usermodifiermanifest.xml does not exist

08:14:03,855 WARN  [mekhq.campaign.stratcon.StratconFacilityManifest] {AWT-EventQueue-0}
mekhq.campaign.stratcon.StratconFacilityManifest.deserialize(StratconFacilityManifest.java:49) - Specified file ./data/stratconfacilities/u
[mekhq.log](https://github.com/MegaMek/mekhq/files/9020074/mekhq.log)
serfacilitymanifest.xml does not exist
`

**Comment by IllianiCBT** [comment]
Can confirm issue is present as of .47.9 #804

Looks to be an identical error in the log
```08:18:49,349 WARN  [mekhq.campaign.stratcon.StratconFacilityManifest] {AWT-EventQueue-0}
mekhq.campaign.stratcon.StratconFacilityManifest.deserialize(StratconFacilityManifest.java:53) - Specified file ./data/stratconfacilities/userfacilitymanifest.xml does not exist

08:18:51,056 WARN  [mekhq.campaign.mission.atb.AtBScenarioManifest] {AWT-EventQueue-0}
mekhq.campaign.mission.atb.AtBScenarioManifest.Deserialize(AtBScenarioManifest.java:53) - Specified file ./data/scenariotemplates/UserScenarioManifest.xml does not exist

08:18:51,217 WARN  [mekhq.campaign.mission.atb.ScenarioModifierManifest] {AWT-EventQueue-0}
mekhq.campaign.mission.atb.ScenarioModifierManifest.Deserialize(AtBScenarioModifier.java:622) - Specified file ./data/scenariomodifiers/usermodifiermanifest.xml does not exist```

@mjbroekman I don't suppose you can remember what you've edited. For example, have you made any changes to the mission folder at all? Or is it just the normal, vanilla, edits of new skins and/or units?

**Comment by mjbroekman** [comment]
I have made no changes to the scenario or mission folders. The only changes I have in this campaign are custom units.

**Comment by Thom293** [comment]
Im pretty sure this is present in the latest nightly version, and as far as I know, happens every single time a lance is deployed to the map "blind."  Or at least it happens often enough I have ceased to think it was odd.  I just assumed it was the intended behavior.

**Comment by mjbroekman** [comment]
Yeah. This is less about the behavior itself and more about the disparity in dates between the Daily Report, the Briefing Room, and the Track map.

**Comment by JonhyBurner** [comment]
> Im pretty sure this is present in the latest nightly version, and as far as I know, happens most times a lance is deployed to the map "blind." Or at least it happens often enough I have ceased to think it was odd. I just assumed it was the intended behavior.

I was browsing my earliest active tickets and saw #2708, so this happens since the beginning of StratCon, so no one experienced this in any other way, even if it is odd. Not a top priority, but some consistency with dates would be beneficial.
